### ISSUE TITLE
Add OCaml context for Live Templates

### ISSUE THREAD

**Issue by QuentinRa** [giraud/reasonml-idea-plugin#342]
I'm a student learning OCaml, and I'm using your plugin in a particular way, as [you can see/read here](https://github.com/memorize-code/jetbrains-ocaml-template). I defined some **Live Templates**, but I can set the scope so that they are only available in my OCaml file.

After checking a bit, it seems **really easy**, but that's up to you to add it, and not me.

The official tutorial is here: [Adding Live Templates to a Plugin](https://plugins.jetbrains.com/docs/intellij/template-support.html), but it seems the section "Implement TemplateContextType" is exactly what I need.

Do you think it would be possible to add it?

**Comment by giraud** [comment]
yes, totally.
I'm also interested by your blockers in intellij wrt ocaml.

**Comment by giraud** [comment]
by the way, I think that if you copy/code the RmlContextType class it should be relatively easy to create a PR

**Comment by QuentinRa** [comment]
Thanks 👍!

Are you asking, what did I had a hard time with? If you do, here are my notes to set up [OCaml in IntelliJ with dune on Windows](https://github.com/lgs-games/memorize/blob/88476ddeff3525a460f8cba9486199609689d134/info/ocaml/intro/install-idea.md) (my grammar is poor, but it got deleted, so I won't improve it, sorry).

**It was working**, but this was not something that I enjoyed using 😅.

I tried esy too, I can look for my notes If you want, but this was still not enjoyable (=running commands, or too many things to install, and the installer for Windows being discontinued, this didn't look good enough for me).
